libs/core/src/lib/dialog/dialog-footer-button/dialog-footer-button.component.ts

```diff
@@ -2,16 +2,14 @@ import { Component } from '@angular/core';
 import { BarElementDirective } from '../../bar/directives/bar-element.directive';
 
 /**
- * Applies fundamental layout and styling to the contents of a dialog body.
+ * Building block of the dialog used to create message box button.
```

Comment:
Yes, this sucks. I think perfect solution would be putting buttons and stuff into the footer like:
```
<footer>
    <fd-button></fd-button>
    <fd-button></fd-button>
<footer>
```
and some general styling coming from the footer would add the spacing between footer direct children.
The problem is we need to use BarComponent for the implementation of the dialog header and footer. BarComponent enforces following structure:
```
<div fd-bar>
    <div fd-bar-right>
        <fd-bar-element>
            <fd-button></fd-button>
        </fd-bar-element>
        <fd-bar-element>
            <fd-button></fd-button>
        </fd-bar-element>
    </div>
</div>   
``` 
so every bar element needs to be wrapped inside `fd-bar-element` which is responsible for creating spacing between bar elements.
The easiest solution is telling the user to wrap footer elements into `fd-bar-element` by himself. To make this more user-friendly I created `fd-dialog-footer-button` which is exactly the same as `fd-bar-element` but with more clear name. However I believe this can and should be simplified, but I wanted to tackle this separately.
Here is a separate issue for this: #3926